packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -104,15 +129,89 @@ export abstract class DeployRetrieveExecutor<
   protected abstract postOperation(
     result: DeployRetrieveResult | undefined
   ): Promise<void>;
+
+  private async handleSourceConflictError(e: any) {
+    const componentPaths = e.data.map(
+      (component: { filePath: any }) => component.filePath
+    );
+    const cacheResult = await MetadataCacheService.loadCacheFor(componentPaths);
+
+    const detector = new TimestampConflictDetector();
+    const diffs = detector.createDiffs(cacheResult, true);
+
+    const conflictMessages = this.getMessagesFor(this.logName);
+    if (conflictMessages) {
+      const conflictChecker = new TimestampConflictChecker(
+        false,
+        conflictMessages
+      );
+      const username = await ConfigUtil.getUsername();
+      await conflictChecker.handleConflicts(
+        componentPaths,
+        String(username),
+        diffs
+      );
+    }
+  }
+
+  private getMessagesFor(
```

Comment:
The SourceConflict error is thrown asynchronously, so in the catch we've lost a handle on which command is being executed or the conflict messages that should be used for each command.  I put them here and am using the logName property to determine the command that is being executed and then which messages to use.